### PR TITLE
Updating unit tests so that they pass with v1beta3 api

### PR DESCRIPTION
**Pull request by nikhiljindal**
Required for #1473 and #5475

This PR fixes all unit tests except pkg/client, which I am still working on. Will fix that in the next PR.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Is the plan to still only support a single testAPI.Version on a single run of tests?

**Comment by nikhiljindal** [comment]
We want to run tests twice once for each version (https://github.com/GoogleCloudPlatform/kubernetes/issues/1473), atleast when run locally.
On travis, we might pick a version at random.

File: pkg/api/testapi/testapi.go [hunk 1]
```diff
@@ -74,3 +75,34 @@ func SelfLink(resource, name string) string {
 	}
 	return fmt.Sprintf("/api/%s/%s/%s", Version(), resource, name)
 }
+
+// Returns the appropriate path for the given resource, namespace and name.
+// For ex, this is of the form:
+// /api/v1beta1/pods/pod0 for v1beta1 and
+// /api/v1beta3/namespaces/foo/pods/pod0 for v1beta3.
+func ResourcePath(resource, namespace, name string) string {
```
Comment (bgrant0607) [inline pkg/api/testapi/testapi.go]: I'm surprised these helpers didn't already exist in the client library itself.

**Comment by bgrant0607** [comment]
There's concern about the time required to run all tests with both API versions on Travis/Shippable.

LGTM.

File: pkg/api/testapi/testapi.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/api/testapi/testapi.go]: Trying to discourage direct path behavior, really

**Pull request merged**